dagshub/models/model_locator.py

```diff
@@ -108,8 +109,10 @@ def _handle_path(self, path: str) -> Tuple[str, StorageType]:
 
         if str_path.startswith("dagshub_storage/"):
             return str_path[len("dagshub_storage/") :], StorageType.DagshubStorage
+        in_path = PurePosixPath(str_path)
         for storage in self.repo_storages:
-            if str_path.startswith(f"{storage.name}"):
+            storage_path = PurePosixPath(storage.name)
+            if is_path_relative_to(in_path, storage_path):
```

Comment:
I don't fully understand how this logic relates to the simpler older one of `startswith` - did you check that it works correctly when the supplied path is exactly the root of the repo? Will `relative_to` return `.` in that case and work correctly?

Reply:
I don't really care about _what_ the relative path exactly is, I just care that in_path is a sub-path of the storage path.

Just in case, went to check this right now:
```
>>> a = PurePosixPath("/")
>>> b = PurePosixPath("/")
>>> a.is_relative_to(b)
True
```

So we should be fine, if a and b is the same it is considered relative